pkg/debuginfo/store.go

```diff
@@ -214,6 +214,10 @@ func (s *Store) upload(ctx context.Context, buildID, hash string, r io.Reader) e
 		if hash != "" && metadataFile != nil {
 			if metadataFile.Hash == hash {
 				level.Debug(s.logger).Log("msg", "debug info already exists", "buildid", buildID)
+				if err := s.metadata.MarkAsUploaded(ctx, buildID, hash); err != nil {
```

Comment:
> There can be confusing circumstances, when for example debuginfos have already been uploaded in the past, but there was no metadata. 

If the metadata was never created, won't this fail all the time?

https://github.com/parca-dev/parca/blob/41a8b3003d8cf411cd57a36ac2c8201456c36112/pkg/debuginfo/metadata.go#L144-L148

Reply:
Good catch!